programs/buck_version.py

```diff
@@ -80,17 +80,13 @@ def get_clean_buck_version(dirpath, allow_dirty=False):
 
 
 def get_dirty_buck_version(dirpath):
-    git_tree_in = check_output(
-        ['git', 'log', '-n1', '--pretty=format:%T', 'HEAD', '--'],
-        cwd=dirpath).strip()
-
     with EmptyTempFile(prefix='buck-git-index') as index_file:
+        subprocess.check_call(
+            ['git', 'read-tree', '--index-output', index_file.name, '--reset', 'HEAD'],
```

Comment:
In this particular case, running with ```--dry-run``` would actually prevent the updated index from being written to ```index_file.name```.  (```--index-output``` prevents our modifications from being written back to the original index file)

(I got to the same point on the man page... had to just poke at it a lot!  If it's worth some explanatory comments in the code, please feel free to request they be added)